packages/data-point/lib/entity-types/base-entity/resolve.js

```diff
@@ -105,13 +107,31 @@ function createCurrentAccumulator (accumulator, reducer, entity) {
     context: entity,
     reducer: currentReducer,
     initialValue: accumulator.value,
-    params: entity.params,
+    params: assignParamsHelper(accumulator, entity),
     debug: debugEntity(reducer.entityType)
   })
 
   return currentAccumulator
 }
 
+/**
+ * Incase there is an override, assigns parameters to the correct requests.
+ * @param {*} accumulator
+ * @param {*} entity
+ */
+function assignParamsHelper (accumulator, entity) {
+  if (
+    !isEmpty(accumulator.overrideEntity) &&
+    accumulator.overrideEntity[entity.entityType]
+  ) {
+    return merge(
+      accumulator.overrideEntity[entity.entityType].params,
+      entity.params
+    )
+  }
+  return entity.params
+}
```

Comment:
Hi @arik223, thanks for tackling this issue! A few suggestions about this function:

1. It should have tests (same for any changes to data-point); if you run `yarn test` locally, it warns you that test coverage has decreased (travis will also give that warning). The goal is to maintain the current level of test coverage.

2. `!isEmpty(accumulator.overrideEntity)` can probably be removed. Since `{}` is the default value for `accumulator.overrideEntity`, it should be safe to do  this:

```js
if (accumulator.overrideEntity[entity.entityType])
```

That checks for the value that will be used, so the more general `isEmpty` is redundant.

3. The `merge` call should probably look like this:

```js
return merge(
  {},
  entity.params,
  accumulator.overrideEntity[entity.entityType].params,
)
```
2 reasons for this:

- If `entity.params` comes last, that will override the overrides, which is not the desired behavior.

- This mutates `accumulator.overrideEntity[entity.entityType].params`, which might lead to "cross contamination" of data. For example, if we do this:

```js
accumulator = assignParamsHelper(accumulator, requestEntity1)
accumulator = assignParamsHelper(accumulator, requestEntity2)
```

The params from `requestEntity1` would be merged into `requestEntity2`, which shouldn't happen.

Reply:
Those are very good points, I have made the necessary changes